workbench/_api/routes/generate.py

```
@@ -0,0 +1,115 @@
+
+def process_generation(prompt_tokens, completion_tokens, tokenizer) -> GenerateData:
+    prompt = tokenizer.batch_decode(prompt_tokens[0])
+    completion = tokenizer.batch_decode(completion_tokens[0])
+
+    return GenerateData(prompt=prompt, completion=completion)
+
+
+@router.post("/start", response_model=GenerateResponse)
+async def start_generate(
+    req: GenerateRequest,
+    state: AppState = Depends(get_state),
```

Ruff (0.16.1)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)